packages/react-router/lib/types.ts

```diff
@@ -66,15 +66,12 @@ type _CreateLoaderData<
   ClientLoaderHydrate extends boolean,
   HasHydrateFallback
 > =
-  [HasHydrateFallback, ClientLoaderHydrate]  extends [true, true] ?
+  [HasHydrateFallback, ClientLoaderHydrate] extends [true, true] ?
     IsDefined<ClientLoaderData> extends true ? ClientLoaderData :
     undefined
   :
   [IsDefined<ClientLoaderData>, IsDefined<ServerLoaderData>] extends [true, true] ? ServerLoaderData | ClientLoaderData :
-  IsDefined<ClientLoaderData> extends true ?
-    ClientLoaderHydrate extends true ? ClientLoaderData :
-    ClientLoaderData | undefined
-  :
+  IsDefined<ClientLoaderData> extends true ? ClientLoaderData :
```

Comment:
If a route has a client loader but no server loader, then `clientLoader.hydrate` is implied and we will always call the loader before rendering the component, so no need for this fork here to include `undefined`

Reply:
Am I understanding this right that the absence of a `loader` and the presence of a `clientLoader` means that the route is automatically CSR only?

I don't think I ever realized that 😅